src/main/java/uk/gov/hmcts/befta/dse/ccd/DataLoaderToDefinitionStore.java

```diff
@@ -315,24 +316,35 @@ protected void importDefinitionsAt(String definitionsPath) {
         List<String> definitionFileResources = getAllDefinitionFilesToLoadAt(definitionsPath);
         logger.info("{} definition files will be uploaded to '{}' on {}.", definitionFileResources.size(),
                 definitionStoreUrl, getDataSetupEnvironment());
+        String message = "Couldn't import {} - Exception: {}.\n\n";
         try {
             for (String fileName : definitionFileResources) {
                 try {
                     logger.info("\n\nImporting {}...", fileName);
                     importDefinition(fileName);
                     logger.info("\nImported {}.\n\n", fileName);
-                } catch (Exception e) {
-                    logger.error("Couldn't import {} - Exception: {}.\n\n", fileName, e);
-                    if (!shouldTolerateDataSetupFailure()) {
-                        throw new RuntimeException(e);
+                } catch (ImportException e) {
+                    logger.error(message, fileName, e);
+                    if (!shouldTolerateDataSetupFailure(e)) {
```

Comment:
we can avoid this new method with throwable parameter when we actually don't have any logic

Reply:
We have that logic in civils. That's why we made this change.

Basically we noticed that while importing an HTTP-504 can be thrown but in that case the import succeeds (it's just too many imports being processed) so we want to ignore the HTTP-504. What we don't want to ignore is all other errors.

As a result we added a `shouldTolerateDataSetupFailure` method override that accepts one argument being our exception, so we can check the actual HTTP status code.

Adding this overload is the whole point of our change.

Reply:
along with 504 (which is a common behaviour for all service teams), we have to consider other 2xx scenarios as well. So, I am approving this for now to resolve your problem. We will plan to tidy up the http response code handling in the future.